gitclone/git.go

```diff
@@ -280,6 +292,19 @@ func checkout(gitCmd git.Git, arg, branch string, depth int, isTag bool) *step.E
 		}
 		return gitCmd.Fetch(opts...)
 	}); err != nil {
+		if branch != "" {
+			branches, branchesErr := listBranches(gitCmd)
+			if branchesErr == nil {
```

Comment:
We add this logic, to let the user know if wrong branch was provided.
- This means we should check here if the given branch is contained in the `branches` slice, if yes we should not send the new error (`newStepErrorWithRecommendations `)
- We could include this information in the error we return (and the step prints this also), so please update `shortMsg` of the error: something like this: `fmt.Errorf("fetch failed: invalid branch selected: %s, available branches: %s: %v", branch, strings.Join(branches, ", "), err)`, so that both the step will print more informative error message

Reply:
This means that I have to write a `Contains` or `Find` function? What should we do the `branch` is in the `branches` list?

Reply:
Done ;)